src/utils/LoggerManager.h

```diff
@@ -0,0 +1,17 @@
+#ifndef LOGGER_MANAGER_H
+#define LOGGER_MANAGER_H
+
+#include <memory>
+#include <spdlog/spdlog.h>
+
+class LoggerManager {
+public:
+    static void setLogLevel(spdlog::level::level_enum level);
+
+    static std::shared_ptr<spdlog::logger>& getLogger();
+
+private:
+    static std::shared_ptr<spdlog::logger> createLogger();
```

Comment:
use `spdlog::set_default_logger` to set the logger as default globally. Then we can use `spdlog::debug` etc. without calling the logging manager before. This allows for easier usage. spdlog might also do some smart optimizations to avoid the refcounting of the smar_pointer, but i have not checked that. Also `spdlog::get` fullfills the role of a logger manager with the added benefit that we can have different loggers for different purposes. Therefore it feels redundant to have our own implementation.
A `void` function to setup the logger is probably enough.